src/language-html/embed/angular-let-declarations.js

```diff
@@ -0,0 +1,27 @@
+import { group } from "../../document/builders.js";
+import { printAssignmentWithLayout } from "../../language-js/print/assignment.js";
```

Comment:
We don't do this before, and I think we can move this to js printer by doing following:

- create a new `__ng_...` parser
- parse right part, then wrap the right node with a manually created `VariableDeclaration`

Do you think it's reasonable?

Reply:
IMO, the answer is no. I think the current code is simpler.

First, let me confirm one thing. Is your concern that there will be a dependency from the code for HTML formatting to the code for JavaScript?

If so, would moving `printAssignmentWithLayout` to `src/common` solve the your concern? It is just a function that takes `leftDoc`, `operator`, `rightDoc`, and `layout` and returns a new doc, so it is not specific to JavaScript.

Reply:
> Is your concern that there will be a dependency from the code for HTML formatting to the code for JavaScript?

Main concern is about plugin, let me check how `prettier-plugin-pug` works.

Reply:
https://github.com/prettier/plugin-pug/blob/66f1c0f97810cefd667242302c9d6cb7477dffb1/src/printer.ts#L656-L683

It simply call `prettier.format` as I thought.

Reply:
Sorry I still don't quite understand the problem, how does my change affect the plugin?

Reply:
It doesn't, but if we move all print logic to js printer, it will be easier for plugin to use.

Reply:
Alright, I understood the logic wrong, so it's just one line?

```
group([group(leftDoc), operator, group(indent([line, rightDoc]))]);
```

Let's put it in HTML printer (maybe with a comment).

The file size grows by import the file https://github.com/prettier/prettier/actions/runs/9918259561/job/27402696804?pr=16474#step:4:298

Reply:
Sounds good to me

Reply:
https://github.com/prettier/prettier/pull/16474/commits/9b32e578f0cdce421f5d2edd698b01a80183f6ae